web/gallery/Star.lhs

```diff
@@ -18,22 +18,27 @@ height: 400
 To create one quarter of the diagram, we connect the corresponding
 points with headless arrows and alternate the line colors.
 
-> quarter n = mconcat [arrowBetween' 
->   (with & arrowHead .~ noHead
->         & shaftStyle %~ lw thin . lc (colors !! ((xCoord1 p) `mod` 2)))
->   (fst p) (snd p) | p <- ps]
+> quarter :: Int -> Diagram B
+> quarter n = mconcat [ makeLine (getColor i) (p2 (i, 0)) (p2 (0, n - i)) | i <- [0..n] ]
 >   where
->     xCoord1 = round . fst . unp2 . fst
->     ps = zip xs (reverse ys)
->     (xs, ys) = pts n
+>     getColor :: Int -> Colour
+>     getColor i = colors !! (p1 `mod` 2)
+>
+>     makeLine :: Colour -> P2 Double -> P2 Double -> Diagram B
+>     makeLine color p q =
+>       p \~\~ q
```

Comment:
Nothing should need to be escaped, this should be `p ~~ q` rather than `p \~\~ q`.